### ISSUE TITLE
[Feature Request] Autocomplete with ability to add custom item

### ISSUE THREAD

**Issue by sahanxdissanayake** [nextui-org/nextui#2000]
### Is your feature request related to a problem? Please describe.

N/A

### Describe the solution you'd like

As a user I would like to be able to add an item to the autocomplete list if the item is not found

### Describe alternatives you've considered

Alternate library

**Comment by Strapontin** [comment]
Isn't it this feature ?

https://nextui.org/docs/components/autocomplete#custom-value

**Comment by PabloGracia** [comment]
> Isn't it this feature ?
> 
> https://nextui.org/docs/components/autocomplete#custom-value

Not really; the custom value just allows users to write any value they want, but NextUI doesn't provide any way to *add* that custom item to the list of items.

**Comment by sahanxdissanayake** [comment]
I ended up using https://react-select.com/creatable

I will keep this feature request open for now to get more feedback

**Comment by TheOldestJun** [comment]
Wow, I'd really appreciate such kind of functionality! Also using some strange alternatives to achieve this thing!

**Comment by chris-rb2** [comment]
+1 need this, also for multi-select

**Comment by vasujhawar** [comment]
+1

**Comment by jaredliu233** [comment]
+1

**Comment by MrLexisDev** [comment]
+1

**Comment by AzizMenssi** [comment]
This how I managed to make it work , a generic example and I'm using react-hoook-form as well :
`const items = [
  { label: "Option 1" },
  { label: "Option 2" },
  { label: "Option 3" },
];
`
`<Controller
  name="selectedItem"
  control={control}
  render={({ field }) => {
    const [itemList, setItemList] = useState(items);
    const [inputValue, setInputValue] = useState("");

    const handleSelectionChange = (selected) => {
      const selectedItem = itemList.find((item) => item.label === selected);
      if (!selectedItem && selected) {
        const newItem = { label: selected };
        setItemList((prev) => [...prev, newItem]);
        field.onChange(selected);
      } else {
        field.onChange(selected);
      }
    };

    return (
      <div className="col-span-6">
        <Autocomplete
          className="text-black"
          aria-label="Select Item"
          placeholder="Select Item"
          defaultItems={itemList}
          onInputChange={(value) => setInputValue(value)}
          selectedKey={field.value}
          onSelectionChange={handleSelectionChange}
        >
          {(inputValue !== "" &&
            itemList.some((item) =>
              item.label.toLowerCase().includes(inputValue.toLowerCase())
            )) ||
          inputValue === "" ? (
            (item) => (
              <AutocompleteItem className="text-black" key={item.label}>
                {item.label}
              </AutocompleteItem>
            )
          ) : (
            <AutocompleteItem className="text-black" key={inputValue}>
              {inputValue}
            </AutocompleteItem>
          )}
        </Autocomplete>
      </div>
    );
  }}
/>
`
It shows the list of items , and when one item is not included , it allows you to add it to the list , could be improved for each use case
